sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestUtils.scala

```diff
@@ -64,6 +64,7 @@ private[sql] trait SQLTestUtils extends SparkFunSuite with SQLTestUtilsBase with
     if (loadTestDataBeforeTests) {
       loadTestData()
     }
+    SparkSession.setActiveSession(spark)
```

Comment:
The active session should be set before we execute the plan, right? For example, in QueryExecution for each query. What is the reason we need to do it here?

Reply:
The active session is required for instantiating the DataSourceReader, which is done at planning time (spark.readStream.{...}.load()) in order to determine the schema.

Reply:
Discussed offline. What we should do instead is set the default session when the test spark session is initialized, since that initialization doesn't invoke SparkSession.getOrCreate() which normally sets it.

Reply:
(This issue was spun off into #20926.)